### DISCUSSION TITLE
SVG disappearing on some devices

### DISCUSSION THREAD

**Discussion by cgcab** [gilbarbara/react-inlinesvg#230] [Q&A] [answered]
Hello, I didn't create an issue cause it seems more like a strange behavior I am having than a real issue. 

Th problem I am having is that only on some devices (can't tell if it happens on all the devices of the same model but the bug is always there for this 2 devices) I have one svg that I use that doesn't shows on the screen 

I am using this icon in 3 places, one is always working even on those devices 
One on a second screen where I can sometime see it for a few ms and then it disappears (I added a fallback but it doesn't show the fallback meaning it think it's loaded) 
![image](https://github.com/gilbarbara/react-inlinesvg/assets/57679996/c5ef9dda-af56-4281-8baa-f5a4dfccb2a0)
that the second case picture on the left and the right I am using a svg from my cdn and you can see that right is showing and left isn't it should be like this 
![image](https://github.com/gilbarbara/react-inlinesvg/assets/57679996/008333d3-3394-4d43-8b54-81438fc059d6)

And the last case is on another screen where I have exactly the same problem the svg isn't showing. I tried to put another svg and it was working. So should I conclude that the svg is faulty ? if so why is it working on the first case ? 

Did you ever see this kind of behavior ? 

Thanks a lot

**Comment by gilbarbara** [comment]
Hey @cgcab 

Does this issue happen in the same browser on both devices? If the same browser, both have the same version?
It feels like a browser engine parsing it differently, and one of the files might have some weird code in it.
Can you share the faulty SVG code?

**Reply by cgcab** [reply]
Thanks for your answer on both device I am using my code in a webview that is used in a native app. So I don't have the precise version of each browser but I can look for it though should be quite the same I think (I guess a safari as it's ios) 
 
 here is the faulty svg (you need to add a fill to see it if needed ^^)
![calendar_clock](https://github.com/gilbarbara/react-inlinesvg/assets/57679996/619565d4-f3da-4e82-a54c-c9895c4f5e90)

**Comment by gilbarbara** [comment]
Ok. The Safari version is tied with the iOS version so that you can check that.
The SVG code doesn't seem wrong; it opened correctly on all my local browsers (Chrome, Arc, Safari, Firefox).
But maybe try adding a `version` to it, like `version="1.1"`, to see if it has any impact.

**Reply by cgcab** [reply]
Thanks for this I will have a few days without being able to test but I will get back to you as soon as I am back. 
I will explore both of your proposition and tell you about them

**Reply by cgcab** [reply]
Hello and thanks for you patience, after some more testing I found out that this was linked to the iOS 16.1.1 version (I couldn't test all the version to see if other are impacted) and that updating iOS to 17 solved the problem. It's a very strange case to be honest but at least I know the reason now.

**Reply by gilbarbara** [reply, marked as answer]
Hey @cgcab 
Good to know.
Yeah, some browsers/versions parse SVG differently, so they might be stricter on the features you use and the version of the SVG standard. That SVG uses a clipPath to render the icon, but it doesn't really need it; the path alone would work.
